### ISSUE TITLE
Generating reports using Reporter fails (All strings must be XML compatible)

### ISSUE THREAD

**Issue by ArekKuczynski** [damies13/rfswarm#305]
**Describe the bug**
This bug occurs when trying to generate reports when data has characters that are not compatible with XML.
To generate this exception i have used one of the exaple robots from rfswarm documentation. In result there are characters that are not yet supported.
(results from ssh_example.robot file) https://github.com/damies13/rfswarm/blob/master/Doc/AUT_Monitoring.md#unix-linux-aut-example:
```
Robot Framework Swarm: Reporter
        Version 1.4.0
        Configuration File:  X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\RFSwarmReporter.ini
Exception in thread Thread-5 (cg_export_xhtml):
Traceback (most recent call last):
  File "C:\Users\arekk\AppData\Local\Programs\Python\Python39\Lib\threading.py", line 1073, in _bootstrap_inner
    self.run()
  File "C:\Users\arekk\AppData\Local\Programs\Python\Python39\Lib\threading.py", line 1010, in run
    self._target(*self._args, **self._kwargs)
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 9910, in cg_export_xhtml
    core.export_xhtml()
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 3270, in export_xhtml
    self.xhtml_add_sections(body, "TOP", sectionpct)
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 3488, in xhtml_add_sections
    self.xhtml_add_sections(nextparent, sect, sectionpct)
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 3479, in xhtml_add_sections
    self.xhtml_sections_table(thiselmt, id)
  File "X:\Programy\Robot_framework\rfswarm\rfswarm_fork\venv\Lib\site-packages\rfswarm_reporter\rfswarm_reporter.py", line 3868, in xhtml_sections_table
    td.text = str(val)
    ^^^^^^^
  File "src\\lxml\\etree.pyx", line 1043, in lxml.etree._Element.text.__set__
  File "src\\lxml\\apihelpers.pxi", line 749, in lxml.etree._setNodeText
  File "src\\lxml\\apihelpers.pxi", line 737, in lxml.etree._createTextNode
  File "src\\lxml\\apihelpers.pxi", line 1530, in lxml.etree._utf8
ValueError: All strings must be XML compatible: Unicode or ASCII, no NULL bytes or control characters
```

This is the data that is stored in the database. I think the problem is with the square characters which stand for unknown letters.
```
[?2004l
Architektura:             x86_64
  Tryb(y) pracy CPU:      32-bit, 64-bit
  Address sizes:          36 bits physical, 48 bits virtual
  Kolejność bajtów:       Little Endian
CPU:                      2
  Lista aktywnych CPU:    0,1
ID producenta:            GenuineIntel
  Nazwa modelu:           Intel(R) Core(TM)2 Duo CPU     P8600  @ 2.40GHz
    Rodzina CPU:          6
    Model:                23
    Wątków na rdzeń:      1
    Rdzeni na gniazdo:    2
    Gniazd:               1
    Wersja:               10
    BogoMIPS:             4787.75
    Flagi:                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ht tm pbe syscall nx lm con
                          stant_tsc arch_perfmon pebs bts nopl cpuid aperfmperf pni dtes64 monitor ds_cpl smx est tm2 ssse3 cx16 xtpr pdcm sse4_1 xsave lahf_lm 
                          pti dtherm
Caches (sum of all):      
  L1d:                    64 KiB (2 instances)
  L1i:                    64 KiB (2 instances)
  L2:                     3 MiB (1 instance)
NUMA:                     
  Węzłów NUMA:            1
  Procesory węzła NUMA 0: 0,1
Vulnerabilities:          
  Gather data sampling:   Not affected
  Itlb multihit:          KVM: Mitigation: VMX unsupported
  L1tf:                   Mitigation; PTE Inversion
  Mds:                    Vulnerable: Clear CPU buffers attempted, no microcode; SMT disabled
  Meltdown:               Mitigation; PTI
  Mmio stale data:        Unknown: No mitigations
  Retbleed:               Not affected
  Spec rstack overflow:   Not affected
  Spec store bypass:      Vulnerable
  Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
  Spectre v2:             Mitigation; Retpolines, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
  Srbds:                  Not affected
  Tsx async abort:        Not affected
[?2004harek@arek-7439wez:~$

```

**To Reproduce**
1. Open Reporter
2. Load Results and Template
3. Generate HTM/DOCX/XLSX

**Expected behavior**

**Screenshots**
![image](https://github.com/user-attachments/assets/b6b435e9-453e-4fa2-a595-845e35f6259e)

**Desktop (please complete the following information):**
- OS: Linux, macOs
- Reporter Version: v1.4.0 newest 30/9/2024

**Additional context**

**Comment by damies13** [comment]
After doing some investigation looks like what I need to do is just ensure the string is UTF-8 before using it with etree:
- https://stackoverflow.com/questions/14616387/how-do-i-escape-special-character-to-write-xml-using-lxml

- https://docs.python.org/3/library/stdtypes.html#str
As we already use the `str()` function, looks like we just need to add the `, encoding='utf-8'` option here.

**Comment by damies13** [comment]
@ArekKuczynski can you 
1) create a branch for this issue and create a test case to reproduce this (i.e. test cases should be failing)
2) Update rfswarm_reporter.py", line 3868, changing `td.text = str(val)` to `td.text = str(val, encoding='utf-8')` and see if this fixes the issue
3) if fixed check the functions related to generating a html report for `= str(` and add `, encoding='utf-8'` if needed

if fixed then submit the fix as a PR

**Comment by damies13** [comment]
Merged into release branch v1.4.0
